tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CodeownersTools/CodeownersTools.cs

```diff
@@ -0,0 +1,443 @@
+
+                    var validateResult = await ValidateCodeownersEntryForService(
+                        validateRepo ?? "",
+                        validateServiceLabel,
+                        validateRepoPath);
+                    output.Output(validateResult);
+                    return;
+                default:
+                    SetFailure();
+                    output.OutputError($"Unknown command: '{command}'");
+                    return;
+            }
+        }
+
+        [McpServerTool(Name = "UpdateCodeowners"), Description("Adds or deletes codeowners for a given service label or path in a repo.")]
+        public async Task<string> UpdateCodeowners(
+            string repo,
+            string typeSpecProjectRoot,
+            string path = null,
+            string serviceLabel = null,
+            List<string> serviceOwners = null,
+            List<string> sourceOwners = null,
+            bool isAdding = false,
+            string workingBranch = null)
+        {
+            try
+            {
+                // Validate atleast Service Label or Path.
+                if (string.IsNullOrWhiteSpace(serviceLabel) && string.IsNullOrWhiteSpace(path))
+                {
+                    throw new Exception($"Service label: {serviceLabel} and Path: {path} are both invalid. Atleast one must be valid");
+                }
+                // Check if it's management plane.
+                var isMgmtPlane = typespecHelper.IsTypeSpecProjectForMgmtPlane(typeSpecProjectRoot);
+
+                // Normalize service path
+                path = codeownersHelper.NormalizePath(path);
+
+                // Get labels file contents.
+                var labelsFileContent = await githubService.GetContentsSingleAsync(Constants.AZURE_OWNER_PATH, Constants.AZURE_SDK_TOOLS_PATH, Constants.AZURE_COMMON_LABELS_PATH);
+
+                if (labelsFileContent == null)
+                {
+                    throw new Exception("Could not retrieve labels file from the repository.");
+                }
+
+                var labelsContent = labelsFileContent.Content;
+                var labelsSha = labelsFileContent.Sha;
+
+                // Validate service label
+                if (!string.IsNullOrEmpty(serviceLabel))
+                {
+                    var serviceLabelValidationResults = LabelHelper.CheckServiceLabel(labelsContent, serviceLabel);
+                    if (serviceLabelValidationResults != LabelHelper.ServiceLabelStatus.Exists)
+                    {
+                        var labelsPullRequests = await githubService.SearchPullRequestsByTitleAsync(Constants.AZURE_OWNER_PATH, Constants.AZURE_SDK_TOOLS_PATH, "Service Label");
+
+                        if (!LabelHelper.CheckServiceLabelInReview(labelsPullRequests, serviceLabel) && string.IsNullOrEmpty(path))
+                        {
+                            throw new Exception($"Service label: {serviceLabel} is invalid.");
```

Comment:
Thinking out loud: I wonder if this should get prescriptive and say something like "create a service label". Perhaps that would help the agent know what to do next?